### DISCUSSION TITLE
Documentation for DeepSleep

### DISCUSSION THREAD

**Discussion by thomas-lentz** [arendst/Tasmota#12490] [Q&A] [answered]
Hi all,
in the [description for executing commands before entering Deepsleep](https://tasmota.github.io/docs/DeepSleep/#driver-writer-executing-commands-before-entering-deepsleep) it is said:
`... it is a good practice to process the handler FUNF_BEFORE_DEEPSLEEP to enable the low power mode. `
using the Callback in an xsns driver creates an "not found" error. 

I couldn't find `*BEFORE_DEEPSLEEP` in `tasmota.h` and of course I also tried "FUN**C**_BEFORE_DEEPSLEEP" which can't be found in any tasmota file (current version Tasmota 9.5.0 Michael - non-development).

I guess the documentation is not correct here.

Kind regards
 Tom

**Comment by barbudor** [comment, marked as answer]
You are correct, this seems to be obsolete
If you look at the BMP driver, you will see that "before deepsleep" code is executed by
``` C++
#ifdef USE_DEEPSLEEP
      case FUNC_SAVE_BEFORE_RESTART:
        BMP_EnterSleep();
        break;
#endif // USE_DEEPSLEEP
```

**Reply by barbudor** [reply]
I'm updating the doc

**Reply by thomas-lentz** [reply]
Thank you Barbudor.
I found another documentation error - is this (community) the right way to process this? A bug report seems to be a bit exagerated.
Greetings Tom

**Reply by barbudor** [reply]
Tom
You can also directly edit the doc using the pencil icon on top right. 
That will automatically fork the doc repo to your github account where you can modify and then make a PR.
Otherwise, here or Discord are both places to report doc mismatches
